Userland/Libraries/LibCrypto/ASN1/DER.cpp

```diff
@@ -29,21 +29,21 @@ Result<Tag, DecodeError> Decoder::read_tag()
     return Tag { (Kind)kind, (Class)class_, (Type)type };
 }
 
-Result<size_t, DecodeError> Decoder::read_length()
+ErrorOr<size_t> Decoder::read_length()
 {
     auto byte = TRY(read_byte());
     size_t length = byte;
 
     if (byte & 0x80) {
         auto count = byte & 0x7f;
         if (count == 0x7f)
-            return DecodeError::InvalidInputFormat;
+            return Error::from_string_literal("Length has an invalid count value");
```

Comment:
Perhaps most (if not all) of these error should contain `DER: ` or something to that extend.

If a user gets a `Length has an invalid count value` this does not really help I would say.

Reply:
Dynamic Error string could help with this by prepending `DER` for any error but I don't know if that is a pattern we want to go for nor can we at the moment.

Reply:
I am mainly trying to keep error messages as unique as possible for now, so that developers can efficiently grep for it. Making them more accessible to the user I decided to postpone until we decide on whether we want to have dynamic error strings (or even better, backtraces).